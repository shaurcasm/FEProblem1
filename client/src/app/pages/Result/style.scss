@import '../../colors.scss';

.result-container {
    margin: auto;
    color: $cultured;
    text-align: center;
    font-size: 1.5rem;

    #success {
        color: $light-green;
    }

    #fail {
        color: crimson;
    }

    #planet-name {
        color: $pumpkin;
    }
}