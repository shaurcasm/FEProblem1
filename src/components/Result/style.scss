@import '../../constants/colors.scss';

.result-container {
    margin: auto;
    color: $cultured;
    text-align: center;
    font-size: 1.5rem;

    #error {
        color: crimson;
        font-size: 2.1rem;
    }

    #error-message {
        color: $pumpkin;
        font-size: 2.1rem;
    }

    #success {
        color: $light-green;
    }

    #fail {
        color: $pumpkin;
    }

    #planet-name {
        color: $pumpkin;
    }
}