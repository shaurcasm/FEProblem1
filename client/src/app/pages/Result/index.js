import React from 'react';
import PropTypes from 'prop-types';
import './style.scss';

class Result extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            result: {}
        }
    }

    handleResult(result) {
        if(result.error) {
            return <p><span id='error'>Error: </span> <span id='error-message'>{result.error}</span></p>
        }

        else if(result.status === 'success') {
            return <p>It's a <span id='success'>success</span>! Found Queen in <span id='planet-name'>{result["planet_name"]}</span>.</p>
        }

        else {
            return <p>Mission <span id='fail'>failed</span>! We'll get her next time.</p>
        }
    }

    render() {
        const result = this.props.location.state.result;

        return(
            <div className='result-container'>
                We're in the endgame now.{this.handleResult(result)}
                Thank you for playing this Demo.
            </div>
        )
    }
}

Result.propTypes = {
    location: PropTypes.object.isRequired
};

export default Result;