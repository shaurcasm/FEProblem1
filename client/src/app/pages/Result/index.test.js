import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Result from './index.js';

describe('<Result /> Page', () => {

    test('Renders with a success result', () => {

        var mockLocation = {
            state: {
                result: {
                    status: 'success',
                    'planet_name': 'FooBar'
                }
            }
        };
        const component = shallow(<Result location={mockLocation} />)
        expect(toJson(component)).toMatchSnapshot();
    });

    test('Renders with a fail result', () => {

        var mockLocation = {
            state: {
                result: {
                    status: 'false'
                }
            }
        };
        const component = shallow(<Result location={mockLocation} />)
        expect(toJson(component)).toMatchSnapshot();
    });

    test('Renders with a error result', () => {
        var errorMessage = 'Token is a satirical character in South Park.';
        var mockLocation = {
            state: {
                result: {
                    error: errorMessage
                }
            }
        };
        const component = shallow(<Result location={mockLocation} />)
        expect(toJson(component)).toMatchSnapshot();
    });
});