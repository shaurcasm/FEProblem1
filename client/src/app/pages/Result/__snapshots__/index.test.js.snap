// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result /> Page Renders with a error result 1`] = `
<div
  className="result-container"
>
  We're in the endgame now.
  <p>
    <span
      id="error"
    >
      Error: 
    </span>
     
    <span
      id="error-message"
    >
      Token is a satirical character in South Park.
    </span>
  </p>
  Thank you for playing this Demo.
</div>
`;

exports[`<Result /> Page Renders with a fail result 1`] = `
<div
  className="result-container"
>
  We're in the endgame now.
  <p>
    Mission 
    <span
      id="fail"
    >
      failed
    </span>
    ! We'll get her next time.
  </p>
  Thank you for playing this Demo.
</div>
`;

exports[`<Result /> Page Renders with a success result 1`] = `
<div
  className="result-container"
>
  We're in the endgame now.
  <p>
    It's a 
    <span
      id="success"
    >
      success
    </span>
    ! Found Queen in 
    <span
      id="planet-name"
    >
      FooBar
    </span>
    .
  </p>
  Thank you for playing this Demo.
</div>
`;
